packages/jest-validate/src/warnings.js

```diff
@@ -13,17 +13,26 @@
 import type {ValidationOptions} from './types';
 
 const chalk = require('chalk');
-const {format, logValidationWarning, WARNING} = require('./utils');
+const {
+  format,
+  logValidationWarning,
+  createDidYouMeanMessage,
+  WARNING,
+} = require('./utils');
 
 const unknownOptionWarning = (
   config: Object,
+  exampleConfig: Object,
   option: string,
   options: ValidationOptions
 ): void => {
+  const didyoumean =
+    createDidYouMeanMessage(option, Object.keys(exampleConfig));
   /* eslint-disable max-len */
   const message =
-`  Unknown option ${chalk.bold(option)} with value ${chalk.bold(format(config[option]))} was found.
-  This is either a typing error or a user mistake. Fixing it will remove this message.`;
+  `  Unknown option ${chalk.bold(`"${option}"`)} with value ${chalk.bold(format(config[option]))} was found.` +
+  (didyoumean && ` ${didyoumean}`) +
+  `\n  This is either a typing error or a user mistake. Fixing it will remove this message.`;
```

Comment:
I changed it to "This is probably a typo", as _typo_ seems to be a valid dictionary term and is a little more polite than "this is a user mistake"

<img width="423" alt="screen shot 2017-01-19 at 09 06 58" src="https://cloud.githubusercontent.com/assets/5106466/22098460/b44ca7f6-de26-11e6-8211-73667633d0e6.png">